src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/InitializeParameter/CSharpInitializeParameterService.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InitializeParameter;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.InitializeParameter;
+
+using static InitializeParameterHelpersCore;
+
+[ExportLanguageService(typeof(IInitializeParameterService), LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpInitializeParameterService() : AbstractInitializerParameterService<StatementSyntax>
+{
+    public override SyntaxNode GetBody(SyntaxNode methodNode)
```

Comment:
extracted from the refactoring provider, into the lang service.  otherwise unchanged.